### PR TITLE
Enable connection to secure broker endpoints

### LINKED ISSUE #126: Parse error while attempting to connect to druid broker
I'm attempting to connect Turnilo to my druid broker but I get the following Parse Error:
```
turnilo --druid druid-broker132.hostname.com:4443
Starting Turnilo v1.4.2
Adding cluster manager for 'druid' with 0 dataCubes
Turnilo is listening on address :: port 9090
Failed to connect to cluster 'druid' because: Error: Parse Error (will retry in 19859ms)
```
The config generation command fails as well:
```
turnilo --druid druid-broker132.hostname.com:4443  --print-config --with-comments > config.yaml
There was an error generating a config: Could not find any data cubes, please verify network connectivity
```

I have confirmed that I'm able to connect to the broker host from the host from which I'm running Turnilo. Any suggestions on what could be wrong here?

### PR DESCRIPTION
**Pull request by a2l007**
Fixes #126 
Attempting to connect to https enabled broker endpoints fail with ParseError. This is due to the fact that there is no explicit way to specify if the protocol is Plain or TLS.
plywood-druid-requester supports connecting to https endpoints via a `protocol` property. This PR enables the end user to specify the protocol as TLS if the broker is secured and it would be passed while creating the request to the broker.
Tested internally and with the changes it is able to successfully connect to https broker endpoints.

### CODE REVIEW

**Comment by a2l007** [comment]
Working on travis build failure.

File: src/server/utils/requester/requester.ts [hunk 1]
```diff
@@ -56,7 +57,8 @@ export function properRequesterFactory(options: ProperRequesterOptions): Plywood
       requester = druidRequesterFactory({
         host,
         timeout: timeout || 30000,
-        requestDecorator: options.druidRequestDecorator
+        requestDecorator: options.druidRequestDecorator,
+        protocol: options.protocol
```
Comment (adrianmroz) [inline src/server/utils/requester/requester.ts]: `druidRequesterFactory` accepts `DruidRequesterParameters` from `plywood-druid-register`. There, `protocol` field is declared as `export declare type Protocol = 'plain' | 'tls-loose' | 'tls'`

https://github.com/implydata/plywood-druid-requester/blob/master/src/druidRequester.ts#L29

Comment (a2l007) [reply src/server/utils/requester/requester.ts]: Is it sufficient to cast it to the Protocol type?

Comment (adrianmroz) [reply src/server/utils/requester/requester.ts]: Ideally I would define `protocol` field in our code with type `Protocol` from `plywood-druid-register`. They're exporting it. That way we also could define default value in our code (probably to 'plain').

**Comment by a2l007** [comment]
@adrianmroz @mkuthan  Please review.

**Comment by a2l007** [comment]
@adrianmroz Sorry for bumping an old issue, but is this patch available in the latest release? I configured 1.14, but it looks like this patch is not available there.

**Pull request merged**